### DISCUSSION TITLE
IP address by programm

### DISCUSSION THREAD

**Discussion by simonik** [dl9rdz/rdz_ttgo_sonde#283] [TTGO smartphone app]
I use android phone and hotspot to connect  TTGO. How can I find out IP address  in "MIT app invertor" that is allocated by phone ?
Now I have to write it by hand. 
For example, how does the rdzsonde application find out? 

Thanks

**Comment by LukePrior** [comment]
mDNS

Default: http://rdzsonde.local/

**Reply by dl9rdz** [reply]
Exactly, the TTGO makes mDNS announcements.

The app on android uses the NsdManager (https://developer.android.com/reference/android/net/nsd/NsdManager) to discover the IP address, see code in https://github.com/dl9rdz/rdzwx-plugin/blob/main/src/android/rdzwx-a.kt

I don't know if MIT app inventor has support for mDNS discovery (seems not to be the case I guess), but probably you could implement this in a custom extension.
[removed the duplicate post in the welcome announcement]

**Reply by simonik** [reply]
> mDNS
> 
> Default: http://rdzsonde.local/

Thanks. But it is not work on android. Only PC.

**Comment by simonik** [comment]
I have to add on the start loop code (udpAddree = 255.255.255.255,udpPort 3333) :
void loop() {
//moje
 if(connected2){
    //Send a packet
   
  int now = millis();
  if (lastUDPtime + 600 < now){
    Serial.println("Posilam UDP");
    udp.beginPacket(udpAddress,udpPort);
    //udp.printf("Seconds since boot: %lu", WiFi.localIP());
    udp.println(WiFi.localIP());
    udp.endPacket();
    lastUDPtime=millis();
  }
  }
//konec moje
